diff --git a/jsfx/jsfx/jsfx_api.cpp b/jsfx/jsfx/jsfx_api.cppdiff --git a/jsfx/jsfx_api.cpp b/jsfx/jsfx_api.cpp

index 36f7a4d2..3bf4f2f2 100644index 36f7a4d2..3770e653 100644

--- a/jsfx/jsfx/jsfx_api.cpp--- a/jsfx/jsfx_api.cpp

+++ b/jsfx/jsfx/jsfx_api.cpp+++ b/jsfx/jsfx_api.cpp

@@ -16,13 +16,14 @@ const char *g_config_slider_classname = "REAPERhfader";@@ -16,13 +16,14 @@ const char *g_config_slider_classname = "REAPERhfader";

  

 HINSTANCE g_hInst; HINSTANCE g_hInst;

  

-extern "C" {-extern "C" {

--

-#ifdef _WIN32-#ifdef _WIN32

-__declspec(dllexport) jsfxAPI JesusonicAPI=-__declspec(dllexport) jsfxAPI JesusonicAPI=

-#else-#else

-__attribute__((visibility("default"))) jsfxAPI JesusonicAPI=-__attribute__((visibility("default"))) jsfxAPI JesusonicAPI=

-#endif-#endif

+// extern "C" {+// extern "C" {

++

+// #ifdef _WIN32+// #ifdef _WIN32

+// __declspec(dllexport)+// __declspec(dllexport) 

+// #else+// #else

+// __attribute__((visibility("default"))) jsfxAPI JesusonicAPI=+// __attribute__((visibility("default"))) jsfxAPI JesusonicAPI=

+// #endif+// #endif

+jsfxAPI JesusonicAPI =+jsfxAPI JesusonicAPI =

 { {

   JSFX_API_VERSION_CURRENT,   JSFX_API_VERSION_CURRENT,

   

@@ -70,16 +71,15 @@ __attribute__((visibility("default"))) jsfxAPI JesusonicAPI=@@ -70,16 +71,16 @@ __attribute__((visibility("default"))) jsfxAPI JesusonicAPI=

 }; };

  

  

-BOOL WINAPI DllMain(HINSTANCE hDllInst, DWORD fdwReason, LPVOID res)-BOOL WINAPI DllMain(HINSTANCE hDllInst, DWORD fdwReason, LPVOID res)

-{-{

-  if (fdwReason==DLL_PROCESS_ATTACH)-  if (fdwReason==DLL_PROCESS_ATTACH)

-  {-  {

-    g_hInst=hDllInst;-    g_hInst=hDllInst;

-  }-  }

-  return TRUE;-  return TRUE;

-}-}

-+// BOOL WINAPI DllMain(HINSTANCE hDllInst, DWORD fdwReason, LPVOID res)

-};+// {

+// Remove DllMain to avoid conflict with JUCE plugin+//   if (fdwReason==DLL_PROCESS_ATTACH)

+// BOOL WINAPI DllMain(HINSTANCE hDllInst, DWORD fdwReason, LPVOID res)+//   {

+// {+//     g_hInst=hDllInst;

+//   if (fdwReason==DLL_PROCESS_ATTACH)+//   }

+//   {+//   return TRUE;

+//     g_hInst=hDllInst;+// }

+//   } 

+//   return TRUE;-};

+// }+// };

  

 extern int g_last_srate; extern int g_last_srate;

   

diff --git a/jsfx/jsfx/jsfx_api.h b/jsfx/jsfx/jsfx_api.hdiff --git a/jsfx/jsfx_api.h b/jsfx/jsfx_api.h

index 349eaf42..8a4154bb 100644index 349eaf42..2ab1063f 100644

--- a/jsfx/jsfx/jsfx_api.h--- a/jsfx/jsfx_api.h

+++ b/jsfx/jsfx/jsfx_api.h+++ b/jsfx/jsfx_api.h

@@ -18,7 +18,9 @@@@ -5,13 +5,14 @@

   *

 // to get the API struct, GetProcAddress("JesusonicAPI")  * API used by REAPER

   */

-#ifndef _WIN32-

+#ifdef _WIN32+#pragma once

+#include <windows.h> 

+#else #ifndef _JESUSONIC_DLL_H_

 #include "../WDL/swell/swell.h" #define _JESUSONIC_DLL_H_

 #endif 

 #ifndef _JS_SFXUI_H_
-#define SX_Instance void
+// #define SX_Instance void
+#include "sfxui.h"
 #endif
 
 #define JSFX_API_VERSION_CURRENT 0xf0f01091
